annlite/index.py

```diff
@@ -378,24 +384,74 @@ def _search_documents(
 
     def filter(
         self,
-        filter: Dict = {},
+        filter: Dict,
         limit: int = 10,
+        offset: int = 0,
+        order_by: Optional[str] = None,
+        ascending: bool = True,
         include_metadata: bool = True,
     ):
+        """Find the documents by the filter.
 
+        :param filter: the filter to be applied to the search.
+        :param limit: the number of results.
+        :param offset: the offset of the results.
+        :param order_by: the field to order the results.
+        :param ascending: whether to order the results in ascending order.
+        :param include_metadata: whether to return document metadata in response.
+        """
         cells = [x for x in range(self.n_cells)]
-        where_clause, where_params = Filter(filter).parse_where_clause()
+        where_clause, where_params = Filter(filter or {}).parse_where_clause()
 
         match_docs = self.filter_cells(
             cells=cells,
             where_clause=where_clause,
             where_params=where_params,
+            limit=limit,
+            offset=offset,
+            order_by=order_by,
+            ascending=ascending,
             include_metadata=include_metadata,
         )
-        return match_docs[0:limit]
+        if limit > 0:
+            return match_docs[:limit]
+        return match_docs
 
-    def _cell_selection(self, query_np, limit):
+    def get_doc_by_id(self, doc_id: str):
+        """Get the document by id.
+
+        :param doc_id: the document id.
+        """
+
+        return self._get_doc_by_id(doc_id)
+
+    def get_docs(
```

Comment:
I prefer `det_docs_by_filtering`

Reply:
we already have an API `def filter()` which works for filtering (in this API, the `filter` parameter is not optional anymore).